### PR TITLE
Fixes check plumbing admin verb

### LINKED ISSUE #59168: Check Plumbing verb breaks horribly with smart pipes
## Testmerges:
N/A

## Reproduction:

Enable debug verbs from the debug tab, then in the mapping verb run the "Check Plumbing" verb. Your client will freeze for up to five minutes before blowing out the chat buffer with a list of every single mapped atmos piping segment:

image

### PR DESCRIPTION
**Pull request by norill**
## About The Pull Request
Fixes #59168

## Why It's Good For The Game
bugs bad

## Changelog
:cl:
fix: fixed check plumbing admin verb
/:cl:

### CODE REVIEW

**Comment by vincentiusvin** [comment]
I can vouch for this PR, it works very well on the map PR linked above. Thanks a lot.

File: code/modules/admin/verbs/atmosdebug.dm [hunk 1]
```diff
@@ -11,16 +11,24 @@
...
+		if(istype(pipe, /obj/machinery/atmospherics/pipe/smart) || istype(pipe, /obj/machinery/atmospherics/pipe/layer_manifold))
+			continue
 		if(pipe.z && (!pipe.nodes || !pipe.nodes.len || (null in pipe.nodes)))
 			to_chat(usr, "Unconnected [pipe.name] located at [ADMIN_VERBOSEJMP(pipe)]", confidential = TRUE)
 
+	//Nodes
+	for(var/obj/machinery/atmospherics/node1 in GLOB.machines)
+		for(var/obj/machinery/atmospherics/node2 in node1.nodes)
+			var/found = FALSE
+			for(var/obj/machinery/atmospherics/node1maybe in node2.nodes)
```
Comment (AnturK) [inline code/modules/admin/verbs/atmosdebug.dm:25]: You can just `if (node1 in node2.nodes)`

File: code/modules/atmospherics/machinery/pipes/layermanifold.dm [hunk 2]
```diff
@@ -27,16 +27,12 @@
 	return ..()
 
 /obj/machinery/atmospherics/pipe/layer_manifold/proc/nullifyAllNodes()
-	var/list/obj/machinery/atmospherics/needs_nullifying = get_all_connected_nodes()
+	for(var/obj/machinery/atmospherics/A in nodes)
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/pipes/layermanifold.dm:30]: Are these changes somehow related to the debug verb ?

File: code/modules/atmospherics/machinery/pipes/layermanifold.dm [hunk 3]
```diff
@@ -77,18 +73,18 @@
 /obj/machinery/atmospherics/pipe/layer_manifold/proc/findAllConnections()
 	front_nodes = list()
 	back_nodes = list()
-	var/list/new_nodes = list()
+	nodes = list()
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/pipes/layermanifold.dm:76]: I assume the whole type wasn't avoiding using nodes variable for any particular reason ?

Comment (norill) [reply code/modules/atmospherics/machinery/pipes/layermanifold.dm:76]: afaik it wasn't using nodes, because it used it's own front_nodes and back_nodes variables. but that breaks the contract the superclass

File: code/modules/atmospherics/machinery/pipes/layermanifold.dm (hunk 2 above)
Comment (norill) [reply code/modules/atmospherics/machinery/pipes/layermanifold.dm:30]: yes. manifolds didnt populate its nodes list, which resulted in false one-way connection detections
